### ISSUE TITLE
Selecting occurences doesnt work as expected

### ISSUE THREAD

**Issue by adnanjpg** [zed-industries/zed#7495]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

lets take for example:

```
my "example" "" text
```

I move my cursor to the double quote and select it. What I now want is to select all occurrences of double quotes. What I do is to press `Cmd + Shift + L`, but doing that results in selecting the `""` before the `text` word together as a single selection. 

I'm not sure if this is the intended behavior, but this it makes hard working with some texts. A real world example (the one I had this issue with) would be something like:
```
cat /tmp/sshkey.pub | awk '{s=sprintf("%296s","");gsub(/ /,"X",s);print $1,s "+" s,$3}'
```
Trying to select the double quotes here and turning them into single quotes is not possible due to this issue

### Environment

Zed: v0.120.6 (Zed)
OS: macOS 14.2.1
Memory: 8 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

https://github.com/zed-industries/zed/assets/54329101/b796f3b8-ff45-4936-988b-ba80c2a56d35

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by JosephTLyons** [comment]
You are right , we are coalescing adjacent selections. Checking with VS Code and Sublime Text, neither do that - so we are doing something different than expected. Nice catch!
